skipruntime-ts/wasm/src/internals/skipruntime_module.ts

```diff
@@ -63,6 +63,31 @@ abstract class SkFrozen extends Frozen {
   }
 }
 
+export function check<T>(value: T): void {
+  if (
+    typeof value == "string" ||
+    typeof value == "number" ||
+    typeof value == "boolean"
+  ) {
+    return;
+  } else if (typeof value == "object") {
+    if (value === null || isSkFrozen(value)) {
+      return;
+    }
+    if (Object.isFrozen(value)) {
+      if (Array.isArray(value)) {
+        value.forEach(check);
+      } else {
+        Object.values(value).forEach(check);
+      }
+    } else {
+      throw new Error("Invalid object: must be deep-frozen.");
+    }
+  } else {
+    throw new Error(`'${typeof value}' cannot be deep-frozen.`);
```

Comment:
Not specific to this PR but: this else branch corresponds to `bigint`, `undefined`, `null`, and `symbol`, or is there an other type? Maybe here (or in `deepFreeze()` directly) is a good place to add a comment explaining why those can't be frozen.